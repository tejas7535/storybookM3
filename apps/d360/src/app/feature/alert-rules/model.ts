import { formatISO } from 'date-fns';

import { ResponseWithResultMessage } from '../../shared/utils/error-handling';

export interface AlertRuleResponse {
  count: number;
  content: AlertRule[];
}

export interface AlertRule {
  id: string;
  deleteData?: boolean | null;
  salesArea?: string | null;
  salesOrg?: string | null;
  type?: string | null;
  customerNumber?: string | null;
  customerName?: string | null;
  materialNumber?: string | null;
  materialDescription?: string | null;
  materialClassification?: string | null;
  sectorManagement?: string | null;
  productionLine?: string | null;
  productLine?: string | null;
  demandPlannerId?: string | null;
  demandPlanner?: string | null;
  gkamNumber?: string | null;
  gkamName?: string | null;
  startDate: Date | null;
  execDay?: ExecDay | null;
  execInterval?: ExecInterval | null;
  endDate: Date | null;
  deactivated?: boolean | null;
  activeCount?: number | null;
  inactiveCount?: number | null;
  completedCount?: number | null;
  alertComment?: string | null;
  region?: string | null;
  generation?: string | null;
  usernameCreated?: string | null;
  usernameLastChanged?: string | null;
  threshold1?: string | null;
  threshold1Description?: string | null;
  threshold1Type?: string | null;
  threshold2?: string | null;
  threshold2Description?: string | null;
  threshold2Type?: string | null;
  threshold3?: string | null;
  threshold3Description?: string | null;
  threshold3Type?: string | null;
  currency?: string | null;
}

export interface AlertTypeDescription {
  alertType: string;
  defaultName: string;
  description: string;
  generation: string;
  threshold1Description: string;
  threshold1Type: string;
  threshold2Description: string;
  threshold2Type: string;
  threshold3Description: string;
  threshold3Type: string;
}

export const execIntervallValues = [
  'M1',
  'M2',
  'M3',
  'M6',
  'W1',
  'D1',
] as const;
export type ExecInterval = (typeof execIntervallValues)[number];
export const execDayValues = ['M01', 'M15', 'W6', 'D'] as const;
export type ExecDay = (typeof execDayValues)[number];

export type AlertRuleSaveResponse = {
  id?: string | null;
  deleteData?: boolean | null;
  type?: string | null;
  region?: string | null;
  salesArea?: string | null;
  salesOrg?: string | null;
  customerNumber?: string | null;
  customerName?: string | null;
  materialNumber?: string | null;
  materialDescription?: string | null;
  materialClassification?: string | null;
  sectorManagement?: string | null;
  demandPlannerId?: string | null;
  productionLine?: string | null;
  productLine?: string | null;
  gkamNumber?: string | null;
  startDate: string | null;
  execDay?: string | null;
  execInterval?: string | null;
  endDate?: string | null;
  generation?: string | null;
  inactiveCount?: string | null;
  currency?: string | null;
  alertComment?: string | null;
  threshold1?: string | null;
  threshold2?: string | null;
  threshold3?: string | null;
  deactivated?: boolean | null;
} & ResponseWithResultMessage;

export interface AlertRuleSaveRequest {
  id: string;
  deleteData?: boolean | null;
  type?: string | null;
  region?: string | null;
  salesArea?: string | null;
  salesOrg?: string | null;
  customerNumber?: string | null;
  customerName?: string | null;
  materialNumber?: string | null;
  materialDescription?: string | null;
  materialClassification?: string | null;
  sectorManagement?: string | null;
  demandPlannerId?: string | null;
  productionLine?: string | null;
  productLine?: string | null;
  gkamNumber?: string | null;
  startDate?: string | null;
  execDay?: string | null;
  execInterval?: string | null;
  endDate?: string | null;
  generation?: string | null;
  inactiveCount?: string | null;
  currency?: string | null;
  alertComment?: string | null;
  threshold1?: string | null;
  threshold2?: string | null;
  threshold3?: string | null;
  deactivated?: boolean | null;
}

export function dataToAlertRuleRequest(data: AlertRule): AlertRuleSaveRequest {
  const startDate = data.startDate;
  const endDate = data.endDate;

  return {
    id: data.id,
    deleteData: data.deleteData,
    type: data.type,
    region: data.region,
    salesArea: data.salesArea,
    salesOrg: data.salesOrg,
    customerNumber: data.customerNumber,
    customerName: data.customerName,
    materialNumber: data.materialNumber,
    materialDescription: data.materialDescription,
    materialClassification: data.materialClassification,
    sectorManagement: data.sectorManagement,
    demandPlannerId: data.demandPlannerId,
    productionLine: data.productionLine,
    productLine: data.productLine,
    gkamNumber: data.gkamNumber,
    startDate: startDate
      ? formatISO(startDate, {
          representation: 'date',
        })
      : null,
    execDay: data.execDay,
    execInterval: data.execInterval,
    endDate: endDate
      ? formatISO(endDate, {
          representation: 'date',
        })
      : null,
    generation: data.generation,
    currency: data.currency,
    alertComment: data.alertComment,
    threshold1: data.threshold1,
    threshold2: data.threshold2,
    threshold3: data.threshold3,
    deactivated: data.deactivated,
  };
}
