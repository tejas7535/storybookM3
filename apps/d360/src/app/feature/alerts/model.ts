export interface Alert {
  id?: string;
  open: boolean;
  priority: boolean;
  alertPriority: Priority;
  deactivated: boolean;
  customerNumber?: string;
  customerName?: string;
  materialNumber?: string;
  materialDescription?: string;
  type?: AlertCategory;
  createdAt?: string;
  changedAt?: string;
  dueDate?: string;
  materialClassification?: string;
  customerMaterialNumber?: string;
  customerMaterialNumberCount?: number;
  openFunction?: OpenFunction;
  currency?: string;
  threshold1?: number;
  threshold1Description?: string;
  threshold2?: number;
  threshold2Description?: string;
  threshold3?: number;
  threshold3Description?: string;
  thresholdDeviation?: number;
  comment?: string;
  changedBy?: string;
  completedAt?: string;
  completedBy?: string;
  createdBy?: string;
  generation?: string;
  keyAccount?: string;
  regionPlPackage?: string;
  threshold1Type?: string;
  threshold2Type?: string;
  threshold3Type?: string;
  calcThreshold1?: number;
  calcThreshold2?: number;
  calcThreshold3?: number;
}

export enum AlertStatus {
  ACTIVE = 'ACTIVE',
  COMPLETED = 'COMPLETED',
  DEACTIVATED = 'DEACTIVATED',
}

export interface AlertNotificationCount {
  openCritical: number;
  openNonCritical: number;
}
export enum Priority {
  Priority1 = 1,
  Priority2 = 2,
  Priority3 = 3,
}

export enum OpenFunction {
  Validation_Of_Demand = 'VOD',
  Customer_Material_Portfolio = 'CMP',
}

export enum AlertCategory {
  CHKDMP = 'CHKDMP',
  DPLEOR = 'DPLEOR',
  DMPFIN = 'DMPFIN',
  DPOPDL = 'DPOPDL',
  CFPRAO = 'CFPRAO',
  CFSUAO = 'CFSUAO',
  PINWAO = 'PINWAO',
  PINWDP = 'PINWDP',
  PIOCAO = 'PIOCAO',
  PIOCDP = 'PIOCDP',
  PINUAO = 'PINUAO',
  PINUDP = 'PINUDP',
  PONWDP = 'PONWDP',
  IAUSDP = 'IAUSDP',
  IAUSAO = 'IAUSAO',
  ACIADP = 'ACIADP',
  SPNWAO = 'SPNWAO',
  SPSBDP = 'SPSBDP',
  REUDAO = 'REUDAO',
  REUSAO = 'REUSAO',
  RESEAO = 'RESEAO',
  RESEDP = 'RESEDP',
  SPCHDP = 'SPCHDP',
  SPCHAO = 'SPCHAO',
  DEVIFC = 'DEVIFC',
  SINWAO = 'SINWAO',
  NPOSDP = 'NPOSDP',
}

export const alertTypesToActivateToggleViaURL: AlertCategory[] = [
  AlertCategory.RESEDP, // Check substitution
  AlertCategory.REUSAO, // Check substitution, orders for predecessors
  AlertCategory.SINWAO, // Substitution Schaeffler intern activated
  AlertCategory.SPCHAO, // Substitution proposal changed
  AlertCategory.SPCHDP, // Substitution proposal changed
  AlertCategory.SPNWAO, // Check substitution proposal
  AlertCategory.SPSBDP, // Substitution proposal blocked
];
