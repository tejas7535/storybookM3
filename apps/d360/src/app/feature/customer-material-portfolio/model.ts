import { SortModelItem } from 'ag-grid-community';

import { ResponseWithResultMessage } from '../../shared/utils/error-handling';
import {
  DemandCharacteristic,
  MaterialClassification,
} from '../material-customer/model';
import { DemandPlanAdoption } from './cmp-modal-types';

/* Singe Phase In*/
export interface CMPSinglePhaseInRequest {
  customerNumber: string;
  materialNumber: string;
  /**
   * format is iso date string YYYY-MM-DDs
   */
  phaseInDate: string;
  demandCharacteristic: string;
}

export type CMPSinglePhaseInResponse = {
  customerNumber: string;
  materialNumber: string;
} & ResponseWithResultMessage;

/* Single Substitution*/
export type CMPSingleSubstitutionResponse = {
  customerNumber: string;
  materialNumber: string;
  confirmationNeeded: boolean;
} & ResponseWithResultMessage;

/* Bulk Phase In*/
export interface CMPBulkPhaseInRequest {
  customerNumber: string;
  phaseInEntities: CMPBulkPhaseInEntity[];
}

export interface CMPBulkPhaseInEntity {
  materialNumber: string;
  /**
   * format is iso date string YYYY-MM-DDs
   */
  phaseInDate: string;
  demandCharacteristic: string;
}

export interface CMPBulkPhaseInResponse {
  customerNumber: string;
  materialResults: CMPMaterialPhaseInResponse[];
}

export type CMPMaterialPhaseInResponse = {
  materialNumber: string;
} & ResponseWithResultMessage;

export interface CMPErrorMessage {
  materialNumber?: string;
  errorMessage: string;
}

/* Update CMPMaterial*/
export interface CMPWriteRequest {
  customerNumber: string;
  materialNumber: string;
  autoSwitchDate: string | null;
  repDate: string | null;
  portfolioStatus: string;
  demandCharacteristic: string | null;
  successorMaterial: string | null;
  demandPlanAdoption: DemandPlanAdoption | null;
}

export type CMPWriteResponse = {
  customerNumber: string;
  materialNumber: string;
} & ResponseWithResultMessage;

export interface CMPEntry {
  region?: string;
  salesArea?: string;
  sectorManagement?: string;
  salesOrg?: string;
  gkamNumber?: string;
  customerNumber?: string;
  materialNumber?: string;
  materialClassification?: MaterialClassification;
  sector?: string;
  productionSegment?: string;
  alertType?: string;
  materialDescription?: string;
  customerMaterialNumber?: string;
  customerMaterialNumberCount?: number;
  packagingSize?: number;
  productionPlant?: string;
  productionLine?: string;
  demandCharacteristic?: DemandCharacteristic;
  portfolioStatus?: string;
  pfStatusAutoSwitch?: string;
  repDate?: string;
  entryValidFrom?: string;
  stochasticType?: string;
  productLine?: string;
  productLineText?: string;
  deliveryQuantity18Months?: number;
  orderQuantity?: number;
  successorMaterial?: string;
  successorMaterialDescription?: string;
  successorCustomerMaterialNumber?: string;
  successorCustomerMaterialNumberCount?: number;
  successorMaterialPackagingSize?: number;
  successorMaterialClassification?: MaterialClassification;
  successorStochasticType?: string;
  successorProductionPlant?: string;
  successorProductionSegment?: string;
  successorProductionLine?: string;
  successorProductLine?: string;
  successorProductLineText?: string;
  successorSchaefflerMaterial?: string;
  successorSchaefflerMaterialDescription?: string;
  successorSchaefflerMaterialPackagingSize?: number;
  hasChildren?: boolean;
  tlMessageType: string | null;
  tlMessage: string | null;
  tlMessageNumber: number | null;
  tlMessageId: string | null;
  tlMessageV1: string | null;
  tlMessageV2: string | null;
  tlMessageV3: string | null;
  tlMessageV4: string | null;
}

export interface CfcrActionRequest {
  customerNumber: string;
  materialNumber: string;
  successorMaterial: string;
}

export interface CfcrActionResponse {
  cfcrActions: [
    {
      materialNumber: string;
      customerNumber: string;
      successorMaterial: string;
      cfcrAction: DemandPlanAdoption;
      selected: boolean;
    },
  ];
}

export interface CMPRequest {
  startRow: number | undefined;
  endRow: number | undefined;
  sortModel: SortModelItem[];
  selectionFilters: any;
  columnFilters: Record<string, any>;
}

export interface CMPResponse {
  headMaterials: { rows: CMPEntry[]; rowCount: number };
  childOfHeadMaterial: Map<string, CMPEntry[]>;
}
