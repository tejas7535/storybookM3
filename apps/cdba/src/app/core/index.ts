export { CoreModule } from './core.module';
export { RoleGuard } from './guards';
