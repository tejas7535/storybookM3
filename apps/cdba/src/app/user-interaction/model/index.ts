export * from './action.model';
export * from './interaction.model';
export * from './interaction-type.enum';
