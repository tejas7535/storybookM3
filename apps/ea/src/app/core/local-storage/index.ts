export * from './ea-parameters-local-storage-item.model';
export * from './local-storage.service';
