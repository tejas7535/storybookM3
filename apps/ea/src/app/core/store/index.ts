export * from './facades';
export * from './reducers';
export * from './selectors';
export * from './store.module';
