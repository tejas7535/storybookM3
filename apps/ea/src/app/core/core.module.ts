import { HTTP_INTERCEPTORS, HttpClientModule } from '@angular/common/http';
import { NgModule } from '@angular/core';
import {
  ErrorStateMatcher,
  ShowOnDirtyErrorStateMatcher,
} from '@angular/material/core';
import { MatIconRegistry } from '@angular/material/icon';
import { DomSanitizer } from '@angular/platform-browser';

import { environment } from '@ea/environments/environment';
import {
  AVAILABLE_LANGUAGES,
  FALLBACK_LANGUAGE,
  LANGUAGE_STORAGE_KEY,
} from '@ea/shared/constants/language';
import {
  TRANSLOCO_PERSIST_LANG_STORAGE,
  TranslocoPersistLangModule,
} from '@ngneat/transloco-persist-lang';

import { SharedTranslocoModule } from '@schaeffler/transloco';

import { HttpBearinxInterceptor } from './interceptor/http-bearinx.interceptor';
import { HttpCatalogWebApiInterceptor } from './interceptor/http-catalog-web-api.interceptor';
import { HttpCO2UpstreamInterceptor } from './interceptor/http-co2-upstream.interceptor';
import { StoreModule } from './store/store.module';

@NgModule({
  imports: [
    StoreModule,

    HttpClientModule,
    SharedTranslocoModule.forRoot(
      environment.production,
      AVAILABLE_LANGUAGES,
      undefined, // default -> undefined would lead to browser detection
      FALLBACK_LANGUAGE.id,
      LANGUAGE_STORAGE_KEY,
      true,
      !environment.localDev,
      undefined,
      `${environment.assetsPath}/i18n/`
    ),

    TranslocoPersistLangModule.forRoot({
      storageKey: LANGUAGE_STORAGE_KEY,
      storage: {
        provide: TRANSLOCO_PERSIST_LANG_STORAGE,
        useValue: localStorage,
      },
    }),
  ],

  providers: [
    {
      provide: HTTP_INTERCEPTORS,
      useClass: HttpBearinxInterceptor,
      multi: true,
    },
    {
      provide: HTTP_INTERCEPTORS,
      useClass: HttpCatalogWebApiInterceptor,
      multi: true,
    },
    {
      provide: HTTP_INTERCEPTORS,
      useClass: HttpCO2UpstreamInterceptor,
      multi: true,
    },
    { provide: ErrorStateMatcher, useClass: ShowOnDirtyErrorStateMatcher },
  ],

  exports: [StoreModule, SharedTranslocoModule],
})
export class CoreModule {
  public constructor(
    private readonly matIconRegistry: MatIconRegistry,
    private readonly sanitizer: DomSanitizer
  ) {
    this.registerEAIcons();
  }

  public registerEAIcons(): void {
    const iconSet: Record<string, string> = {
      co2: 'icon_CO2.svg',
      airwaves: 'icon_airwaves.svg',
      air: 'icon_air.svg',
      acute: 'icon_acute.svg',
      bolt: 'icon_bolt.svg',
      device_thermostat: 'icon_device_thermostat.svg',
      mop: 'icon_mop.svg',
      calculation: 'icon_calculations.svg',
      friction_load: 'icon_load_frictional_powerloss.svg',
      lubrication_parameters: 'icon_lubrication_parameters.svg',
      rating_life: 'icon_rpm_rating_life.svg',
      water_drop: 'icon_water_drop.svg',
    };
    for (const [name, url] of Object.entries(iconSet)) {
      const setUrl = this.sanitizer.bypassSecurityTrustResourceUrl(
        `${environment.assetsPath}/icons/${url}`
      );
      this.matIconRegistry.addSvgIcon(name, setUrl);
    }
  }
}
