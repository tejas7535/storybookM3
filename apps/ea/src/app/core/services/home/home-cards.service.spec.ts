import { waitForAsync } from '@angular/core/testing';
import { provideRouter } from '@angular/router';

import { firstValueFrom } from 'rxjs';

import { createServiceFactory, SpectatorService } from '@ngneat/spectator/jest';

import { provideTranslocoTestingModule } from '@schaeffler/transloco/testing';

import { Card } from './card.model';
import { HomeCardsService } from './home-cards.service';

const assetsPath = 'assetPath';
jest.mock(
  '@ea/core/services/assets-path-resolver/assets-path-resolver.helper',
  () => ({
    getAssetsPath: jest.fn(() => assetsPath),
  })
);

describe('HomeCardsService', () => {
  let spectator: SpectatorService<HomeCardsService>;
  let service: HomeCardsService;

  const createServiceWithProductionConfig = createServiceFactory({
    service: HomeCardsService,
    imports: [provideTranslocoTestingModule({ en: {} })],
    providers: [provideRouter([])],
  });

  beforeEach(() => {
    spectator = createServiceWithProductionConfig();
    service = spectator.inject(HomeCardsService);
    Object.defineProperty(window, 'open', {
      value: jest.fn(),
    });
  });

  it('should be created', () => {
    expect(service).toBeTruthy();
  });

  describe('when getting home cards', () => {
    let result: Card[];

    beforeAll(async () => {
      result = await firstValueFrom(service.homeCards$);
    });

    it('should get home cards', waitForAsync(() => {
      expect(result).toMatchSnapshot();
    }));

    describe('when performing any link action', () => {
      it('should open url', waitForAsync(() => {
        const expectedUrls = [
          'homePage.cards.greaseApp.externalLink',
          'homePage.cards.mountingManager.externalLink',
          'homePage.cards.catalog.externalLink',
          'homePage.cards.contact.externalLink',
        ];

        result.forEach((card) => card.action());

        expect(window.open).toBeCalledTimes(expectedUrls.length);

        expectedUrls.forEach((url) => {
          expect(window.open).toBeCalledWith(url, '_blank');
        });
      }));
    });
  });

  describe('when getting sustainability card', () => {
    let result: Card;

    beforeAll(async () => {
      result = await firstValueFrom(service.sustainabilityCard$);
    });

    it('should get sustainability card', waitForAsync(() => {
      expect(result).toMatchSnapshot();
    }));

    describe('when performing sustainability action', () => {
      it('should open url', waitForAsync(() => {
        result.action();

        expect(window.open).toBeCalled();

        expect(window.open).toBeCalledWith(
          'homePage.sustainabilityCard.action.url',
          '_blank'
        );
      }));
    });
  });
});
