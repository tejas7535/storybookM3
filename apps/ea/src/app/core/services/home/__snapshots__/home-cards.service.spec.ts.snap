// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeCardsService when getting home cards should get home cards 1`] = `
Array [
  Object {
    "action": [Function],
    "mainTitle": "homePage.cards.greaseApp.title.main",
    "subTitle": "homePage.cards.poweredBy",
    "svgIcon": "grease_app",
  },
  Object {
    "action": [Function],
    "mainTitle": "homePage.cards.mountingManager.title.main",
    "subTitle": "homePage.cards.poweredBy",
    "svgIcon": "mounting_manager",
  },
  Object {
    "action": [Function],
    "imagePath": "assetPath/images/homepage/catalog.jpg",
    "mainTitle": "homePage.cards.catalog.title.main",
  },
  Object {
    "action": [Function],
    "imagePath": "assetPath/images/homepage/contact.jpg",
    "mainTitle": "homePage.cards.contact.title.main",
  },
]
`;

exports[`HomeCardsService when getting sustainability card should get sustainability card 1`] = `
Object {
  "action": [Function],
  "actionTitle": "homePage.sustainabilityCard.action.title",
  "mainTitle": "homePage.sustainabilityCard.title",
}
`;
